src/AWS.Deploy.Common/Exceptions.cs

```diff
@@ -8,171 +8,223 @@
 
 namespace AWS.Deploy.Common
 {
-    public class ProjectFileNotFoundException : Exception
+    public abstract class DeployToolException : Exception
     {
-        public ProjectFileNotFoundException(string projectPath)
-            : base($"A project was not found at the path {projectPath}.")
+        public DeployToolErrorCode ErrorCode { get; set; }
+
+        public DeployToolException(DeployToolErrorCode errorCode, string message, Exception? innerException = null) : base(message, innerException)
         {
-            Path = projectPath;
+            ErrorCode = errorCode;
         }
+    }
 
-        public string Path { get; set; }
+    public enum DeployToolErrorCode
```

Comment:
This seems very fragile error code number system, if every a developer adds an enum value in between, everything after that value will get messed up.

Either we should make then int constants or atleast provide documentation that new values **must** be added at the end only

Reply:
Also, these enums will not be backward compatible.

Reply:
Added int values to enums and casted to string when passing through API. The client now accepts a string and has no knowledge of the enum.